Appsilon/rhino
# Simplify CI into a single workflow

### Changes
Use a single workflow file to run R CMD check, linter and unit tests. This PR also fixes CI breakage due to lintr errors (can be seen e.g. [here](https://github.com/Appsilon/rhino/runs/7212652061?check_suite_focus=true), probably due recent changes in lintr).

### How to test
The CI should pass. See also [here](https://github.com/Appsilon/rhino/runs/7213060213?check_suite_focus=true) for example run with breakages.
